aws_batch/pipeline.py

```diff
@@ -357,6 +357,40 @@ def generate_taxid_annotated_m8(input_m8, output_m8, accession2taxid_db):
             outf.write(new_line)
     outf.close()
 
+def filter_deuterostomes_from_m8(input_m8, output_m8, deuterostome_file):
+    with open(deuterostome_file) as f:
+        content = f.readlines()
+    taxids_toremove = [x.strip("\n") for x in content]
+    output_f = open(output_m8, 'wb')
+    with open(input_m8, "rb") as input_f:
+        for line in input_f:
+            taxid = (line.split("taxid"))[1].split(":")[0]
+            if not taxid in taxids_toremove:
+                output_f.write(line)
+    output_f.close()
+
+def filter_taxids_from_fasta(input_fa, output_fa, annotation_prefix, accession2taxid_path, deuterostome_file):
+    with open(deuterostome_file) as f:
+        content = f.readlines()
+    taxids_toremove = [x.strip("\n") for x in content]
+    accession2taxid_dict = shelve.open(accession2taxid_path)
+    input_f = open(input_fa, 'rb')
+    output_f = open(output_fa, 'wb')
+    sequence_name = input_f.readline()
+    sequence_data = input_f.readline()
+    while len(sequence_name) > 0 and len(sequence_data) > 0:
+        read_id = sequence_name.rstrip().lstrip('>')
+        accession_id = (read_id.split(annotation_prefix+":"))[1].split(":")[0] if read_id.startswith(annotation_prefix) else (read_id.split(":"+annotation_prefix+":"))[1].split(":")[0]
```

Comment:
line too long. also i think more explanation will be desirable with an example annotated read_id